File: DocumentAnalysisSystem/Readers/DocxDocumentReader.cs
using System.IO;
using DocumentFormat.OpenXml.Packaging;
using DocumentAnalysisSystem.Models;
using WordDocument = DocumentFormat.OpenXml.Wordprocessing.Document;
using ParagraphElement = DocumentFormat.OpenXml.Wordprocessing.Paragraph;

namespace DocumentAnalysisSystem.Readers
{
    public class DocxDocumentReader : IDocumentReader
    {
        public Document ReadDocument(string filePath)
        {
            var document = new Document(Path.GetFileName(filePath));

            using var wordDocument = WordprocessingDocument.Open(filePath, false);
            var body = wordDocument.MainDocumentPart?.Document.Body;
            if (body == null)
            {
                return document;
            }

            foreach (var paragraph in body.Elements<ParagraphElement>())
            {
                var text = paragraph.InnerText?.Trim();
                if (!string.IsNullOrEmpty(text))
                {
                    document.Add(new Paragraph(text));
                }
            }

            return document;
        }
    }
}

File: DocumentAnalysisSystemJava/src/main/java/com/documentanalysis/readers/DocxDocumentReader.java
package com.documentanalysis.readers;

import com.documentanalysis.models.Document;
import com.documentanalysis.models.Paragraph;
import org.apache.poi.xwpf.usermodel.XWPFDocument;
import org.apache.poi.xwpf.usermodel.XWPFParagraph;

import java.io.FileInputStream;
import java.nio.file.Paths;

public class DocxDocumentReader implements IDocumentReader {
    @Override
    public Document readDocument(String filePath) throws Exception {
        Document document = new Document(Paths.get(filePath).getFileName().toString());

        try (FileInputStream fis = new FileInputStream(filePath);
             XWPFDocument xwpfDocument = new XWPFDocument(fis)) {
            for (XWPFParagraph paragraph : xwpfDocument.getParagraphs()) {
                String text = paragraph.getText();
                if (text != null) {
                    text = text.trim();
                }
                if (text != null && !text.isEmpty()) {
                    document.add(new Paragraph(text));
                }
            }
        }

        return document;
    }
}
